libraries/Networking/src/ESP_Network_Interface.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "esp_netif_types.h"
+#include "esp_event.h"
+#include "Arduino.h"
+#include "ESP_Network_Manager.h"
+
+typedef enum {
+    ESP_NETIF_ID_STA,
+    ESP_NETIF_ID_AP,
+    ESP_NETIF_ID_PPP,
+    ESP_NETIF_ID_ETH0,
+    ESP_NETIF_ID_ETH1,
+    ESP_NETIF_ID_ETH2,
+    ESP_NETIF_ID_MAX
+} ESP_Network_Interface_ID;
+
+#define ESP_NETIF_ID_ETH ESP_NETIF_ID_ETH0
+
+class ESP_Network_Interface {
+    public:
+        ESP_Network_Interface();
+        virtual ~ESP_Network_Interface();
+
+        bool config(IPAddress local_ip = (uint32_t)0x00000000, IPAddress gateway = (uint32_t)0x00000000, IPAddress subnet = (uint32_t)0x00000000, IPAddress dns1 = (uint32_t)0x00000000, IPAddress dns2 = (uint32_t)0x00000000, IPAddress dns3 = (uint32_t)0x00000000);
+
+        const char * getHostname();
+        bool setHostname(const char * hostname);
+
+        virtual bool started() = 0;
+        virtual bool connected() = 0;
+        bool linkUp();
+        bool hasIP();
+        const char * ifkey();
+        const char * desc();
+        String impl_name();
+
+        uint8_t * macAddress(uint8_t* mac);
+        String macAddress();
+        IPAddress localIP();
+        IPAddress subnetMask();
+        IPAddress gatewayIP();
+        IPAddress dnsIP(uint8_t dns_no = 0);
+        IPAddress broadcastIP();
+        IPAddress networkID();
+        uint8_t subnetCIDR();
+        IPAddress localIPv6();
```

Comment:
As mentioned elsewhere, this should be `linkScopeIPv6()` or `linkLocalIPv6()`, because it has a different result than the IPv4 `localIP()`.  

Usually `localIP()` returns a local (as opposed to remote) IPv4 address with global scope, wheres this function returns link-local scope instead.